neuro_extras/main.py

```diff
@@ -474,19 +431,107 @@ async def _nonstorage_cp(
         ]
     else:
         raise ValueError("Unknown cloud provider")
-    click.echo(f"Running {command} {' '.join(args)}")
+    click.echo(f"Running '{command} {' '.join(args)}'")
     subprocess = await asyncio.create_subprocess_exec(command, *args)
     returncode = await subprocess.wait()
+    if UrlType.get_type(source_url) == UrlType.LOCAL and remove_source:
+        _rm_by_url(source_url)
     if returncode != 0:
         raise click.ClickException("Cloud copy failed")
-    elif UrlType.get_type(source_url) == UrlType.LOCAL:
-        source_path = Path(source_url.path)
-        if remove_source:
-            if source_path.is_dir():
-                dir_util.remove_tree(str(source_path))
-            else:
-                if source_path.exists():
-                    source_path.unlink()
+
+
+async def _extract(source_url: URL, destination_url: URL, rm_src: bool) -> None:
+    file = Path(source_url.path)
+    if file.is_dir():
+        file = list(file.glob("*"))[0]
+    suffixes = file.suffixes
```

Comment:
minor: BTW: in future, we better add more logigng on archive type detection. This is an empirical algorithm and thus it should not be silent.